testsuites/CBLTester/System_test_multiple_device/test_system_testing.py

```diff
@@ -0,0 +1,283 @@
+            t.join()
+            query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
+
+        ###########################
+        # Deleting doc on SG side #
+        ###########################
+        docs_to_delete = set(random.sample(doc_ids, num_of_doc_to_delete))
+        sg_docs = sg_client.get_bulk_docs(url=sg_url, db=sg_db, doc_ids=list(docs_to_delete), auth=session)[0]
+        print "Deleting {} docs on SG - {}".format(len(docs_to_delete),
+                                                   docs_to_delete)
+        sg_client.delete_bulk_docs(url=sg_url, db=sg_db,
+                                   docs=sg_docs, auth=session)
+        for repl_obj, repl, cbl_db, query in zip(replicator_obj_list,
+                                                 replicator_list,
+                                                 cbl_db_list,
+                                                 query_obj_list):
+            t = Thread(target=_replicaton_status_check, args=(repl_obj, repl))
+            t.start()
+            t.join()
+            query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
+            time.sleep(5)
+        _check_doc_count(db_obj_list, cbl_db_list)
+        # removing ids of deleted doc from the list
+        doc_ids = doc_ids - docs_to_delete
+
+        ############################
+        # Deleting doc on CBL side #
+        ############################
+        docs_to_delete = set(random.sample(doc_ids, num_of_doc_to_delete))
+        docs_to_delete_per_db = len(docs_to_delete) / len(db_obj_list)
+        i = 0
+        for db_obj, cbl_db, repl_obj, repl, query in zip(db_obj_list,
+                                                         cbl_db_list,
+                                                         replicator_obj_list,
+                                                         replicator_list,
+                                                         query_obj_list):
+            print "deleting {} docs from {} db - {}".format(docs_to_delete_per_db,
+                                                            db_obj.getName(cbl_db),
+                                                            list(docs_to_delete)[i: i + docs_to_delete_per_db])
+            db_obj.delete_bulk_docs(cbl_db, list(docs_to_delete)[i: i + docs_to_delete_per_db])
+            i += docs_to_delete_per_db
+            time.sleep(5)
+            query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
+
+            # Deleting docs will affect all dbs as they are synced with SG.
+            for repl_obj, repl, cbl_db, query in zip(replicator_obj_list,
+                                                     replicator_list,
+                                                     cbl_db_list,
+                                                     query_obj_list):
+                t = Thread(target=_replicaton_status_check, args=(repl_obj, repl))
```

Comment:
as you have start and join together for one thread, it runs only one thead, right?

Reply:
answered above